cli/module_loader.rs

```diff
@@ -21,13 +47,210 @@ use deno_core::ModuleType;
 use deno_core::OpState;
 use deno_core::ResolutionKind;
 use deno_core::SourceMapGetter;
+use deno_graph::source::Resolver;
 use deno_graph::EsmModule;
 use deno_graph::JsonModule;
+use deno_graph::Module;
+use deno_graph::Resolution;
+use deno_lockfile::Lockfile;
+use deno_runtime::deno_node::NodeResolutionMode;
 use deno_runtime::permissions::PermissionsContainer;
+use deno_semver::npm::NpmPackageReqReference;
+use std::borrow::Cow;
 use std::cell::RefCell;
+use std::collections::HashSet;
 use std::pin::Pin;
 use std::rc::Rc;
 use std::str;
+use std::sync::Arc;
+
+pub struct ModuleLoadPreparer {
```

Comment:
Extracted from `ProcState`.

There are a lot of dependencies needed for this, but I don't think it's a big deal because there's only ever one of these created and it will be reduced once I extract out stuff like type checking to separate structs.